### PR TITLE
Find a better way to determine last status change

### LINKED ISSUE #1636: New TO server field: LastStatusChange
As a user, I want to know when the last status change of a server was so that I can track how long a server has been in a given status such as OFFLINE or ADMIN_DOWN.  The only current way to maybe do it is to parse the entire changelog, but that doesn't work with API changes where the individual changes aren't posted to the changelog.

### PR DESCRIPTION
**Pull request by srijeet0406**
## What does this PR (Pull Request) do?

- [x] This PR fixes #1636  

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Ops
- Traffic Portal
- CI tests

## What is the best way to verify this PR?

Run local TO and TP and create a server. Notice that the `status_last_updated` field gets set to the current time in the DB.
Now, go to TP, and select the `Status Last Updated` as one of the display columns for the servers table. Notice that it shows the correct time elapsed since you created the server.
Now, update this server changing any field, EXCEPT the `status` of the server. Notice that the `status_last_updated` value DOES NOT change.
Now, update the server by changing the server's status (you will have to do this from the backend call, since TP doesn't let you do it from the front end). 
Notice that the `status_last_updated` time gets changed to the current time this time around.

Also, make sure unit tests and API tests pass.

## If this is a bug fix, what versions of Traffic Control are affected?

- master

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR
- [x] This PR does not include documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by mitchell852** [comment]
sorry, conflict in our favorite file :)

**Comment by srijeet0406** [comment]
@mitchell852 fixed 👍

File: traffic_ops/traffic_ops_golang/server/servers.go [hunk 1]
```diff
@@ -102,9 +102,102 @@ SELECT
 	s.type AS server_type_id,
 	s.upd_pending AS upd_pending,
 	s.xmpp_id,
-	s.xmpp_passwd
+	s.xmpp_passwd,
+    s.status_last_updated
```
Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/server/servers.go:106]: format problem?

Comment (srijeet0406) [reply traffic_ops/traffic_ops_golang/server/servers.go:106]: hmm, I ran `go fmt` on it 🤔

**Comment by mitchell852** [comment]
for some reason the column filter is not working in TP on this field:

image

I have a server i just changed the status of and it's not showing up. So i'm wondering if the data format needs to be changed to that of LastUpdated:

lastUpdated: "2020-08-28 20:53:55+00",
statusLastUpdated: "2020-08-28T20:53:55.75928Z"

I'm not sure if that will solve the TP grid problem but maybe consistent is better anyhow?

**Comment by ocket8888** [comment]
The data will compare properly if you coerce it to a Date. If you look in the servers table controller, when I set `$scope.servers` I map the returned servers array and apply a conversion to Date objects on the Last Updated field. So no, it's not that the Last Updated format works (in fact it's worse than the statusLastUpdated field, I had to do some string manipulation to get the `Date` constructor to work), it's just that it can only filter Dates by date.

**Comment by srijeet0406** [comment]
Resolved in `97bca2f`

**Comment by mitchell852** [comment]
Can you update `dateCellFormatter` as well to this?

```
	/**
	 * Formats the contents of the 'lastUpdated' and 'statusLastUpdated' column cells as "relative to now".
	 */
	function dateCellFormatter(params) {
		return params.value ? dateUtils.getRelativeTime(params.value) : params.value;
	}
```

so `dateUtils.getRelativeTime` doesn't try to parse a null which displays "Invalid Date"

**Pull request merged**